src/pages/DataAccessRequestApplication_new.js

```diff
@@ -215,6 +230,8 @@ class DataAccessRequestApplicationNew extends Component {
       prev.formData = merge(prev.formData, formData);
       return prev;
     });
+
+    window.addEventListener('scroll', this.onScroll);
```

Comment:
Shoot, i didnt realize codacy wouldnt like this... I don't know how to do this part of the ticket without adding an event listener though. :[ Let me brainstorm

Reply:
If codacy is giving us false positives here, we can change that. I'm not sure what the best practice is, but codacy shouldn't be the primary roadblock.

Reply:
TIL you can bypass codacy warnings by adding disabling eslint on the same lines - would this be okay? XD I guess I'll let the reviewers weigh in hehe

Reply:
There's nothing wrong with your function. It's a textbook definition of a debounce with a reset when the function is called repeatedly. 

Codacy's issue is that it's not context aware of its warnings. Generally `setTimeout` is frowned upon for anything regarding data since defining a fixed time delay on async operations (where length is indeterminate) can be dangerous. In this context though the UI changes are independent of anything else going on in the component, so it's not a problem.

If you care about Codacy warnings, try looking into `lodash's` (vanilla, not fp) `debounce` function and see if that does anything (though odds are it'll be doing the same things under the hood).